clients/egui/src/account/mod.rs

```diff
@@ -217,8 +230,14 @@ impl AccountScreen {
                         }
                     }
                 }
-                AccountUpdate::FileDeleted(f) => self.tree.remove(&f),
-                AccountUpdate::SyncUpdate(update) => self.process_sync_update(ctx, update),
+                AccountUpdate::FileDeleted(f) => {
+                    self.tree.remove(&f);
+                    self.suggested = SuggestedDocs::new(&self.core);
```

Comment:
it's kinda weird to reset it to a new suggested docs imo. I think it's also part of the reason you're having a weird error handling experience. I think you probably want SuggestedDocs to take a context and the tx and have a function that requests a recalc. The recalc ideally happens on a different thread, updates the state of SuggestedDocs and requests a new frame from egui.

Reply:
is a separate thread necessary considering that suggested docs call is sub 20ms if i recall correctly (at least for my case, not a lot of files).